leasing/models/rent.py

```diff
@@ -1476,6 +1476,118 @@ class LegacyIndex(models.Model):
     )
 
 
+class OldDwellingsInHousingCompaniesPriceIndex(models.Model):
+    """
+    In Finnish: Vanhojen osakeasuntojen hintaindeksi
+
+    From my understanding, a unique price index is identified by the API
+    database table URL and code of the index. A single table can hold multiple
+    indexes, and a single code can be used in multiple tables.
+
+    Members:
+        code: Code for the index's table column. Example: "ketj_P_QA_T". Same code for different intervals, e.g. yearly or quarterly.
+        name: Name of the index. Example: "Index (2020=100)".
+        comment: Comment for the index's table column.
+        source: Source of the data.
+        source_table_updated: UTC timestamp when the source table was last updated.
+        source_table_label: Label for the source table.
+        url: API endpoint URL.
+    """
+
+    # Maximum lengths are arbitrary, but set to avoid extra large input.
+    CHARFIELD_MAX_LENGTH = 255
+
+    code = models.CharField(
+        verbose_name=_("Code"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    name = models.CharField(
+        verbose_name=_("Name"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    comment = models.TextField(verbose_name=_("Region"), blank=True)
+    source = models.CharField(
+        verbose_name=_("Data source"),
+        blank=True,
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    source_table_updated = models.DateTimeField(
+        verbose_name=_("Source table updated"), null=True
+    )
+    source_table_label = models.TextField(
+        verbose_name=_("Source table label"),
+        blank=True,
+    )
+    url = models.CharField(
+        verbose_name=_("URL"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+
+    class Meta:
+        verbose_name = pgettext_lazy(
+            "model name", "price index of old dwellings in housing companies"
+        )
+        verbose_name_plural = pgettext_lazy(
+            "model name", "price indexes of old dwellings in housing companies"
+        )
+        constraints = [
+            models.UniqueConstraint(fields=["code", "url"], name="unique_price_index")
```

Comment:
Just checking: code is not unique?

Reply:
`code` appears to be unique for the same index (in this case the `2020=100` price index for old dwellings in housing companies), but it is not unique for the resolution of the index: the same code is used for yearly and quarterly tables. These two tables have different values for other properties of that index column, because one table has the column property values in Finnish and the other in English.

For example, the source table column property `text` (which I save as the index `name` in the model) for the index column has value `Indeksi (2020=100)` in the quarterly table and `Index (2020=100)` in the yearly table. If both tables were imported, the `name` field in the index column for this `code` would churn between the two values unnecessarily. The same is true for the index column property `comment`, which is in Finnish in the quarterly table and in English in the yearly table, and which I save in the index model as `comment`.

This is not a problem right now, because we only import the quarterly table. I anticipate this churning/overwriting issue in the future, if Tilastokeskus adds more indexes in this same table for future years (2025, 2030...) and if those index columns have different values for the `text` or `comment` properties. Or if we later decide to import both the yearly and quarterly index values with this importer.

Reply:
I thought this through once again, and the code must be unique for every index. Otherwise a single table couldn't show multiple different indexes with same "topic", such as we have now. The quarterly/yearly interval is also not a problem because we can sidestep that on our side, e.g. by leaving the quarter value as null if we want to signify a yearly index number.

--> the only unique field needs to be "code".

Reply:
Unique constraint updated to be only `code`